backend/baseclass.pm

```diff
@@ -889,20 +890,25 @@ sub wait_serial {
     }
 
     $regexp = [$regexp] if ref $regexp ne 'ARRAY';
-    my $initial_time = time;
-    while (time < $initial_time + $timeout) {
+    my $initial_time   = time;
+    my $current_offset = $self->{serial_offset};
+  MATCH: while (time < $initial_time + $timeout) {
         $str = $self->serial_text();
         for my $r (@$regexp) {
-            $matched = ref $r eq 'Regexp' ? $str =~ $r : $str =~ m/$r/;
-            if ($matched) {
-                $regexp = "$r";
-                last;
+            if (!$args->{no_regex} && $str =~ m/$r/) {
+                $current_offset += $LAST_MATCH_END[0];
+                $matched = 1;
+                last MATCH;
+            }
+            if ($args->{no_regex} && (my $i = index($str, $r)) >= 0) {
```

Comment:
```suggestion
            } 
            elsif ($args->{no_regex} && (my $i = index($str, $r)) >= 0) {
```

Reply:
I had it initially like that however, I had no useful usage for `else`. Any ideas ? :)

Reply:
that we do not run the `if` condition again if the first match. It's just eyecandy.

Reply:
yes, ofc it is better. I just prefer complete `if -> elsif -> else` syntax and not skipping the `else`. As I had no use for `else` I chose 2x `if`. :)